Core/Transaction.cs

```diff
@@ -0,0 +1,192 @@
+            Output = output;
+            Amount = amount;
+            IsDelegating = isDelegating;
+            Signature = signature;
+		}
+
+        /// <summary>
+        /// Optional GUID, Does not influence verification
+        /// </summary>
+        public Guid Id { get; init; }
+        /// <summary>
+        /// Version of the Transaction (format)
+        /// </summary>
+        public int Version { get; init; }
+        /// <summary>
+        /// Exact creation time of the Transaction
+        /// </summary>
+        public DateTimeOffset CreationTime { get; init; }
+        /// <summary>
+        /// The signature/hash of the previous transaction
+        /// </summary>
+        public byte[] MerkleHash { get; init; }
+        /// <summary>
+        /// The public key of the input/sender of the transaction
+        /// </summary>
+        public byte[] Input { get; init; }
+        /// <summary>
+        /// The public key of the output/reciever of the transaction
+        /// </summary>
+        public byte[] Output { get; init; }
+        /// <summary>
+        /// The amount to be send in the Transaction
+        /// </summary>
+        public int Amount { get; init; }
+        /// <summary>
+        /// If the Transaction is delegating
+        /// </summary>
+        public bool IsDelegating { get; init; }
+        /// <summary>
+        /// The signature/hash of the Transaction
+        /// </summary>
+        public byte[]? Signature { get; set; }
+
+        private byte[]? byteArrayCache;
+
+        /// <summary>
+        /// Convert all data in the Transaction to a ByteArray. This should be Unique for each transaction
+        /// </summary>
+        /// <returns>a ByteArray based on the data in the transaction</returns>
+        public byte[] GetSignatureByteArray() {
+            if (byteArrayCache == null) {
+                byte[] versionByteArray = BitConverter.GetBytes(Version);
+                byte[] timeByteArray = BitConverter.GetBytes(CreationTime.ToUnixTimeSeconds());
+                byte[] amountByteArray = BitConverter.GetBytes(Amount);
+                byte[] delegateByteArray = BitConverter.GetBytes(IsDelegating);
+
+                if (!BitConverter.IsLittleEndian) {
+                    Array.Reverse(versionByteArray);
+                    Array.Reverse(timeByteArray);
+                    Array.Reverse(amountByteArray);
+                    Array.Reverse(delegateByteArray);
+                }
+
+                byteArrayCache = Utility.ConcatArrays(versionByteArray, timeByteArray, MerkleHash, Input, Output, amountByteArray, delegateByteArray);
+            }
+
+            return byteArrayCache;
+        }
+
+        /// <summary>
+        /// Sign the Transaction
+        /// </summary>
+        /// <param name="key">The key used to sign the Transaction</param>
+        /// <exception cref="ArgumentException">Fired if the wrong key is used</exception>
+        public void Sign(ECDsaKey key) {
```

Comment:
Is het de bedoeling dat de manier van signen die we hiervoor gebruikten wordt vervangen hierdoor?